src/SignalRServiceExtension/TriggerBindings/ServerlessHub.cs

```diff
@@ -66,12 +73,26 @@ protected ServerlessHub(IServiceHubContext hubContext = null, IServiceManager se
         /// <summary>
         /// Return a <see cref="SignalRConnectionInfo"/> to finish a client negotiation.
         /// </summary>
-        protected SignalRConnectionInfo Negotiate(string userId = null, IList<Claim> claims = null, TimeSpan? lifeTime = null)
+        protected SignalRConnectionInfo Negotiate(string userId = null, IList<Claim> claims = null, TimeSpan? lifeTime = null, HttpContext httpContext = null)
         {
+            var negotiateResponse = _serviceContext.GetClientEndpointAsync(HubName, httpContext, userId, claims, lifeTime).Result;
             return new SignalRConnectionInfo
             {
-                Url = _serviceManager.GetClientEndpoint(HubName),
-                AccessToken = _serviceManager.GenerateClientAccessToken(HubName, userId, claims, lifeTime)
+                Url = negotiateResponse.Url,
+                AccessToken = negotiateResponse.AccessToken
+            };
+        }
+
+        /// <summary>
+        /// Return a <see cref="SignalRConnectionInfo"/> to finish a client negotiation.
+        /// </summary>
+        protected async Task<SignalRConnectionInfo> NegotiateAsync(string userId = null, IList<Claim> claims = null, TimeSpan? lifeTime = null, HttpContext httpContext = null)
```

Comment:
>NegotiateAsync [](start = 52, length = 14)

why add duplicate method?

Reply:
This is an async method which would be useful for AAD access key.